tiledb/common/thread_pool.h

```diff
@@ -48,6 +48,49 @@
 namespace tiledb {
 namespace common {
 
+/**
+ * A class-scope singleton kept in existence by members, allocated on first
+ * use.
+ */
+template <class T>
+struct class_singleton {
```

Comment:
- use a `class` here instead of a struct
- name all struct/classes similar to `ExistingClasses`
- this does not appear to be tightly coupled with the thread pool, can we put this in its own header file for reuse?

Reply:
It's "struct" to avoid "class { public: ...". Haven't fully documented it yet, but it's designed to be used as a class member with no private members.

Reply:
> can we put this in its own header file for reuse?
That's where it's headed, yes. I hadn't gotten that far in the refactoring. When I pull it out into its own file, it needs to get a unit test suite. I haven't written one yet.

Reply:
NameChanged